src/api/utils/index.ts

```diff
@@ -0,0 +1,20 @@
+export const windowLoaded = new Promise((accept, reject) => {
+  if (typeof window === 'undefined') {
+    return accept()
+  }
+
+  if (typeof window.addEventListener !== 'function') {
+    return reject(new Error('expected to be able to register event listener'))
+  }
+
+  window.addEventListener('load', function loadHandler(event) {
+    window.removeEventListener('load', loadHandler, false)
+    return accept(event)
+  }, false)
+})
+
+
+export const promisify = (func: Function, context: object, ...defArgs: any[]) =>
```

Comment:
I hate this function 😂 😂 😂 😂

Reply:
I'm open to any changes. This one is actually a slightly modified `util.promisify` from node, but we can't use node system libraries in the browser(

Reply:
its fine for now